pykafka/balancedconsumer.py

```diff
@@ -504,18 +515,24 @@ def _check_held_partitions(self):
             self._rebalance()
 
     def _brokers_changed(self, brokers):
+        if not self._running:
+            return False  # disables the watcher
```

Comment:
`ChildrenWatch` explicitly checks for `False`: https://github.com/python-zk/kazoo/blob/bbf581e3db992e22b82f1264b1a21308595699fa/kazoo/recipe/watchers.py#L334

It's really weird though, if you look at that function - I don't think `return False` kills the watcher thread, it just sets it `_stopped`.